src/tls/connection.c

```diff
@@ -624,6 +669,7 @@ connection_handshake (Connection *self)
           return false;
         }
 
+      gnutls_session_set_verify_function (self->tls, verify_peer_certificate);
```

Comment:
Nice! I wonder why I didn't do that back then .. perhaps we still had to support a too old GnuTLS version?